ui/de-lib/src/main/java/org/iplantc/de/client/services/AppUserServiceFacade.java

```diff
@@ -82,4 +82,12 @@ void rateApp(App app,
                  AsyncCallback<AppFeedback> callback);
 
     void deleteRating(App app, AsyncCallback<AppFeedback> callback);
+
+    void getPermissions(List<App> apps, AsyncCallback<String> callback);
+
+    void shareApp(JSONObject request, AsyncCallback<String> callback);
+    
+    void unshareApp(JSONObject request, AsyncCallback<String> callback);
```

Comment:
This refactor should also be applied to `DiskResourceServiceFacade.shareDiskResource(...)` and `DiskResourceServiceFacade.unshareDiskResource(...)`
Change the signatures to;
`void shareDiskResource(Map<String>, List<Sharing>> userSharingMap, AsyncCallback<Void> callback);`
`void unshareDiskResource(Map<String>, List<Sharing>> userUnsharingMap, AsyncCallback<Void> callback);`